lmp/infer/_top_p.py

```diff
@@ -0,0 +1,294 @@
+            # Next token id tensor  : Sampled token id from top K.
+            #                         Use sampled index to get sampled token
+            #                         id from top K next token id tensor.
+            # Next token id shape   : `(1, 1)`.
+            # Next token id dtype   : `torch.int64`.
+            batch_next_tkid_cand_idx = torch.multinomial(
+                batch_topp_tkid_probs,
+                num_samples=1,
+            )
+            batch_next_tkid = torch.gather(
+                batch_topp_tkid,
+                -1,
+                batch_next_tkid_cand_idx,
+            )
+
+            # Concate the last next token id prediction result with previous
+            # token ids prediction result and use to perform further
+            # prediction.
+            # `batch_prev_tkids` shape: `(1, S)`.
+            # `batch_prev_tkids` dtype: `torch.int64`.
+            # `batch_next_tkid`  shape: `(1, 1)`.
+            # `batch_next_tkid`  dtype: `torch.int64`.
+            # Output shape            : `(1, S+1)`.
+            # Output dtype            : `torch.int64`.
+            batch_prev_tkids = torch.cat(
+                [batch_prev_tkids, batch_next_tkid],
+                dim=-1
+            )
+
+            # If the prediction token id is `[eos]`, then stop prediction.
+            if batch_next_tkid[0, 0].item() == tknzr.eos_tkid:
+                break
+
+        # Output generated text.
+        return tknzr.batch_dec(
+            batch_tkids=batch_prev_tkids.tolist(),
+            rm_sp_tks=True,
+        )[0]
+
+    @staticmethod
+    def infer_parser(parser: argparse.ArgumentParser) -> None:
+        r"""Top ``P`` inference method CLI arguments parser.
+
+        Parameters
+        ==========
+        parser: argparse.ArgumentParser
+            Parser for CLI arguments.
+
+        See Also
+        ========
+        lmp.script.generate_text
+            Generate text using pre-trained language model.
+
+        Examples
+        ========
+        >>> import argparse
+        >>> from lmp.infer import TopKInfer
+        >>> parser = argparse.ArgumentParser()
+        >>> TopKInfer.infer_parser(parser)
+        >>> args = parser.parse_args([
+        ...     '--ckpt', '5000',
+        ...     '--exp_name', 'my_exp',
+        ...     '--p', '.9',
+        ...     '--txt', 'Hello world',
+        ... ])
+        >>> args.ckpt == 5000
+        True
+        >>> args.exp_name == 'my_exp'
+        True
+        >>> args.p == .9
+        True
+        >>> args.txt == 'Hello world'
+        True
+        >>> args.seed == 42
+        True
+        """
+        # Load common arguments.
+        BaseInfer.infer_parser(parser=parser)
+
+        # Required arguments.
+        group = parser.add_argument_group('inference method arguments')
+        group.add_argument(
+            '--p',
+            help='Sample from token ids with p probabilities mass.',
```

Comment:
change "Sample from token ids with p probabilities mass." to "Sample from token ids with top p probabilities mass."